resip/src/main/java/fr/gouv/vitam/tools/resip/app/ResipApp.java

```diff
@@ -377,44 +391,64 @@ else if (cmd.hasOption("listimport")) {
                 Instant start, end;
 
                 start = Instant.now();
-                SEDALibProgressLogger spl = new SEDALibProgressLogger(ResipLogger.getGlobalLogger().getLogger(), logLevel);
+                SEDALibProgressLogger spl =
+                    new SEDALibProgressLogger(ResipLogger.getGlobalLogger().getLogger(), logLevel);
 
                 if (creationContext instanceof DiskImportContext) {
                     DiskImportContext diskImportContext = (DiskImportContext) creationContext;
-                    DiskToArchiveTransferImporter di = new DiskToArchiveTransferImporter(diskImportContext.getOnDiskInput(), spl);
+                    DiskToArchiveTransferImporter di =
+                        new DiskToArchiveTransferImporter(diskImportContext.getOnDiskInput(), spl);
                     for (String ip : diskImportContext.getIgnorePatternList())
                         di.addIgnorePattern(ip);
                     di.doImport();
                     packet = di.getArchiveTransfer();
                 } else //noinspection ConstantConditions
                     if (creationContext instanceof SIPImportContext) {
                         SIPImportContext sipImportContext = (SIPImportContext) creationContext;
-                        String target = sipImportContext.getWorkDir() + File.separator + Paths.get(sipImportContext.getOnDiskInput()).getFileName().toString() + "-tmpdir";
-                        SIPToArchiveTransferImporter si = new SIPToArchiveTransferImporter(sipImportContext.getOnDiskInput(), target, spl);
+                        String target = sipImportContext.getWorkDir() + File.separator +
+                            Paths.get(sipImportContext.getOnDiskInput()).getFileName().toString() + "-tmpdir";
+                        SIPToArchiveTransferImporter si =
+                            new SIPToArchiveTransferImporter(sipImportContext.getOnDiskInput(), target, spl);
                         si.doImport();
                         packet = si.getArchiveTransfer();
+                    } else if (creationContext instanceof CSVImportContext) {
+                        CSVMetadataImportContext cSVMetadataImportContext = (CSVMetadataImportContext) creationContext;
+                        CSVMetadataToDataObjectPackageImporter cmi = new CSVMetadataToDataObjectPackageImporter(
+                            cSVMetadataImportContext.getOnDiskInput(), cSVMetadataImportContext.getCsvCharsetName(),
+                            cSVMetadataImportContext.getDelimiter(), spl);
+                        cmi.doImport();
+
+                        packet = new ArchiveTransfer();
+                        Work work = new Work(cmi.getDataObjectPackage(), cSVMetadataImportContext,
+                            new ExportContext(Prefs.getInstance()));
+                        work.getDataObjectPackage()
```

Comment:
Il faut pas contrôler que work.getDataObjectPackage() est toujours non nul ??

Reply:
it is always not null because we initialize it in the constructor
`Work(DataObjectPackage dataObjectPackage, CreationContext creationContext, ExportContext exportContext)`

Reply:
Indeed, the point is to check "cmi.getDataObjectPackage() " that is not null! It is not harmeful to use it in the constructor of work, but it surely is when getting it from "work" ( In the constructor of work, we assign cmi.getDataObjectPackage() to dataObjectPackage ).